brigade/plugins/tasks/networking/netmiko_tasks.py

```diff
@@ -0,0 +1,180 @@
+        parameters["host"] = netmiko_host
+
+    if netmiko_dict is not None:
+        parameters.update(netmiko_dict)
+    else:
+        netmiko_dict = {}
+    parameters['port'] = netmiko_dict.get('port', brigade_host.ssh_port)
+
+    if device_type is None:
+        if netmiko_dict.get("device_type") is None:
+            device_type = brigade_host.nos
+        else:
+            device_type = netmiko_dict.get("device_type")
+
+    # Convert to netmiko device_type format (if napalm format is used)
+    parameters['device_type'] = napalm_to_netmiko_map.get(device_type, device_type)
+    return parameters
+
+
+@netmiko_get_connection
+def netmiko_run(task, method, ip=None, host=None, username=None, password=None,
+                device_type=None, netmiko_dict=None, cmd_args=None, cmd_kwargs=None,
+                netmiko_conn=None):
+    """
+    Execute any Netmiko method from connection class (BaseConnection class and children).
+
+    Arguments:
+        method(str): Netmiko method to use
+        ip (string, optional): defaults to ``brigade_ip``
+        host (string, optional): defaults to None
+        username (string, optional): defaults to ``brigade_username``
+        password (string, optional): defaults to ``brigade_password``
+        device_type (string, optional): Netmiko device_type to use, defaults to ``nos`` (mapped \
+        through napalm_to_netmiko_map)
+        netmiko_dict (dict, optional): Additional arguments to pass to Netmiko ConnectHandler, \
+        defaults to None
+
+    Returns:
+        :obj:`brigade.core.task.Result`:
+          * result (``dict``): dictionary with the result of the getter
+    """
+    # Netmiko connection automatically created by 'netmiko_get_connection' decorator (if needed)
+    netmiko_method = getattr(netmiko_conn, method)
+    if cmd_args is None:
+        cmd_args = ()
+    if cmd_kwargs is None:
+        cmd_kwargs = {}
+    result = netmiko_method(*cmd_args, **cmd_kwargs)
+    return Result(host=task.host, result=result)
+
+
+@netmiko_get_connection
+def netmiko_send_command(task, ip=None, host=None, username=None, password=None,
+                         device_type=None, netmiko_dict=None, use_textfsm=False,
+                         cmd_args=None, cmd_kwargs=None, netmiko_conn=None):
+
+    brigade_host = task.host
+    parameters = netmiko_args(host=brigade_host, ip=ip, netmiko_host=host, username=username,
+                              password=password, device_type=device_type, netmiko_dict=netmiko_dict)
+    if cmd_args is None:
+        cmd_args = ()
+    elif isinstance(cmd_args, py23_compat.string_types):
+        cmd_args = (cmd_args,)
+
+    if cmd_kwargs is None:
+        cmd_kwargs = {}
+
+    if len(cmd_args) >= 1:
+        command = cmd_args[0]
+    else:
+        command = cmd_kwargs['command_string']
+
+    # Netmiko connection automatically created by 'netmiko_get_connection' decorator (if needed)
+    result = netmiko_conn.send_command(*cmd_args, **cmd_kwargs)
```

Comment:
This part still feels very strange to me. 99% of the time I just want to send in a command using send_command. I think there should be one very simple task that just does that.

Reply:
I will probably fix for `netmiko_send_command` as given most of the arguments are going to go away, the reason I wanted to do it is no longer there. Note, this doesn't apply to `netmiko_run_command` which I still think makes sense to do it as I originally did. Which I will expand on below.